credentialdigger/client_sqlite.py

```diff
@@ -398,6 +499,38 @@ def get_discovery_group(self, repo_url, state=None):
                 WHERE repo_url=? GROUP BY file_name, snippet, state'
                                            )
 
+    def get_embedding(self, discovery_id=None, snippet=None):
+        """ Retrieve a discovery embedding.
+
+        Parameters
+        ----------
+        discovery_id: int
+            The id of the discovery whose embedding is being retrieved
+        snippet: str
+            The snippet whose embedding is being retrieved. Only used if
+            discovery_id is not provided
+
+        Returns
+        ------
+        list
+            The embedding
+        """
+
+        if discovery_id:
+            query = 'SELECT embedding FROM embeddings WHERE id=?'
+        elif snippet:
+            query = 'SELECT embedding FROM embeddings WHERE snippet=?'
+        else:
+            return None
+        str_embedding = super().get_embedding(query=query,
+                                              discovery_id=discovery_id,
+                                              snippet=snippet)
+        embedding = []
+        if str_embedding:
+            embedding = [float(emb)
+                         for emb in str_embedding[0].split(",")[:-1]]
```

Comment:
wasn't `str_embedding` a string?
why `str_embedding[0].split(...)`? it will split 1 letter

Reply:
Actually there is a mistake in the docstring of get_embedding: str_embedding is a tuple containing a single string, not a string

Reply:
I will move the [0] to the parent definition, so that get_embedding does indeed return a string/list, but as is, the code works